src/msg/Messenger.cc

```diff
@@ -21,7 +21,7 @@ Messenger *Messenger::create_client_messenger(CephContext *cct, std::string lnam
 uint64_t Messenger::get_pid_nonce()
 {
   uint64_t nonce = getpid();
-  if (nonce == 1) {
+  if (nonce <= 10 || getenv("CEPH_CONTAINER_HAS_INIT")) {
```

Comment:
```suggestion
  if (nonce <= 10 || getenv("CONTAINER_IMAGE")) {
```

(nit) rook and cephadm currently set `CONTAINER_IMAGE` ... maybe that would simply things?

See: https://github.com/rook/rook/blob/master/pkg/operator/k8sutil/pod.go#L301

Reply:
updated.  also switched it back to pid == 1

Reply:
should we always set CEPH_USE_RANDOM_NONCE (regardless of if an init is used)?